test/powershell/Modules/Microsoft.PowerShell.Management/FileSystem.Tests.ps1

```diff
@@ -0,0 +1,230 @@
+    }
+
+    AfterAll {
+        #restore the previous location
+        Set-Location -Path $restoreLocation
+    }
+
+    BeforeEach {
+        Set-Location -Path "TestDrive:\"
+    }
+
+    It "Verify Get-Location and Set-Location" {
+        $currentLoc = Get-Location
+        Set-Location $level1_0
+        $level1Loc = Get-Location
+        Set-Location $level2_0
+        $level2Loc = Get-Location
+        $currentLoc.Path | Should Be (Join-Path "TestDrive:" "")
+        $level1Loc.Path | Should Be (Join-Path "TestDrive:" $level1_0)
+        $level2Loc.Path | Should Be (Join-Path (Join-Path "TestDrive:" $level1_0) $level2_0)
+    }
+
+    It "Verify Push-Location and Pop-Location" {
+        #push a bunch of locations
+        Push-Location
+        $push0 = Get-Location
+        Set-Location $level1_0
+        Push-Location
+        $push1 = Get-Location
+        Set-Location $level2_0
+        Push-Location
+        $push2 = Get-Location
+
+        #navigate back home to change path out of all pushed locations
+        Set-Location "TestDrive:\"
+
+        #pop locations off
+        Pop-Location
+        $pop0 = Get-Location
+        Pop-Location
+        $pop1 = Get-Location
+        Pop-Location
+        $pop2 = Get-Location
+
+        $pop0.Path | Should Be $push2.Path
+        $pop1.Path | Should Be $push1.Path
+        $pop2.Path | Should Be $push0.Path
+    }
+}
+
+Describe "Validate basic Content Cmdlets for the FileSystem provider" -Tags "CI" {
+    BeforeAll {
+        $testFile = "testFile.txt"
+        $restoreLocation = Get-Location
+        Set-Location "TestDrive:\"
+    }
+
+    AfterAll {
+        #restore the previous location
+        Set-Location -Path $restoreLocation
+    }
+
+    BeforeEach {
+        New-Item -Path $testFile -ItemType File
+    }
+
+    AfterEach {
+        Remove-Item -Path $testFile -Force -ErrorAction SilentlyContinue
+    }
+
+    It "Set-Content to a file" {
+        $content =  Set-Content -Value "some content" -Path $testFile -PassThru
+        $content | Should BeExactly "some content"
+    } 
+
+    It "Add-Content to a file" {
+        $content = Set-Content -Value "some content" -Path $testFile -PassThru
+        $addContent = Add-Content -Value " more content" -Path $testFile -PassThru
+        $fullContent = Get-Content -Path $testFile
+        $content | Should Match "some content"
+        $addContent | Should Match "more content"
+        ($fullContent[0] + $fullContent[1]) | Should Match "some content more content"
+    }
+
+    It "Clear-Content of a file" {
+        Set-Content -Value "some content" -Path $testFile
+        $contentBefore = Get-Content -Path $testFile
+        Clear-Content -Path $testFile
+        $contentAfter = Get-Content -Path $testFile
+        $contentBefore.Count | Should Be 1
+        $contentAfter.Count | Should Be 0
+    }
+}
+
+Describe "Validate Resolve-Path Cmdlet Parameters" -Tags "CI" {
```

Comment:
Once all the code for the provider PR's has been accepted. I will go back and rename all the describes to be more consistent as a new PR. It will be clearer to see what I have done than to update all 5 of my provider PR's separately right now.